src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -93,8 +108,28 @@ def create_connectedk8s(cmd, client, resource_group_name, cluster_name, https_pr
     # if the user had not logged in.
     check_kube_connection(configuration)
 
+    required_node_exists = check_linux_amd64_node(configuration)
+    if not required_node_exists:
+        telemetry.set_user_fault()
+        telemetry.set_exception(fault_type=consts.Linux_Amd64_Node_Not_Exists,
+                                summary="Couldn't find any node on the kubernetes cluster with the architecture type 'amd64' and OS 'linux'")
+        logger.warning("Please ensure that this Kubernetes cluster have any nodes with OS 'linux' and architecture 'amd64', for scheduling the Arc-Agents onto and connecting to Azure. Learn more at {}".format("https://aka.ms/ArcK8sSupportedOSArchitecture"))
+
     # Get kubernetes cluster info
     kubernetes_version = get_server_version(configuration)
+    current_version_details = kubernetes_version.split('.')  # Format v1.20.0S
+    current_version_minor = current_version_details[1]
+    current_version_major = current_version_details[0][1:]
+    try:
+        latest_kubernetes_version = utils.get_latest_kubernetes_version()
+        latest_version_details = latest_kubernetes_version.split('.')
+        latest_version_minor = latest_version_details[1]
+        latest_version_major = latest_version_details[0][1:]
+        if (int(current_version_major) < int(latest_version_major)) or (int(current_version_minor) < int(latest_version_minor) - 2):
+            logger.warning("The Kubernetes cluster you are trying to connect to Azure Arc is of version {}, which is out of Azure Arc's support window. Learn more at {}".format(kubernetes_version, "https://aka.ms/ArcK8sVersionSupportPolicy"))
```

Comment:
For 1.20.0 - major= 1, minor=20, patch = 0
For 1.19.4 - major = 1, minor= 19, patch = 4

If the kubernetes cluster being onboarded is v1.19.4, then it wont get the warning, since its minor(19) is not less than latest release's minor -2(20-2=18).

Let me know if you meant something else.

Reply:
The latest version is v1.20.4, and my aks cluster has kubernetes version v1.18.6, so it doesnt get the warning based on the conditioning i have put and as shashank had explained.

Reply:
Yeah, I think I gave the wrong example. Lets say the k8s version is 1.22.0, Next if the k8s version released is 2.1.0 then if someone still uses 1.22.0, you would return error. as major version of current is smaller.. However with our n-2 logic, 1.22.0 and 1.21.0 must be supported (last two releases) right? 

---
In reply to: [597053069](https://github.com/Azure/azure-cli-extensions/pull/3001#discussion_r597053069) [](ancestors = 597053069)

Reply:
The n-2 check is supposed to be only for minor version. That major check is added because otherwise we cant validate properly. Latest: v2.0.0, current: v1.20.6, this minor check would be passing. Confirming with shashank once.

Reply:
The n-2 check is for last two releases and not just minor releases and this is where this check fails. I feel such things are best documented.

Reply:
Also, we are not erroring out, this is warning that k8s latest > current

Reply:
Removed check